modules/secrets-store-aws.adoc

```diff
@@ -0,0 +1,370 @@
+// Module included in the following assemblies:
+//
+// * nodes/pods/nodes-pods-secrets-store.adoc
+//
+// IMPORTANT: This file requires you to define :secrets-store-provider: before including this module.
+
+ifeval::["{secrets-store-provider}" == "AWS Secrets Manager"]
+:aws-secrets-manager:
+endif::[]
+ifeval::["{secrets-store-provider}" == "AWS Systems Manager Parameter Store"]
+:aws-systems-manager-parameter-store:
+endif::[]
+
+:_content-type: PROCEDURE
+[id="secrets-store-aws_{context}"]
+= Mounting secrets from {secrets-store-provider}
+
+You can use the {secrets-store-operator} to mount secrets from {secrets-store-provider} to a CSI volume in {product-title}. To mount secrets from {secrets-store-provider}, your cluster must be installed on AWS and use AWS Security Token Service (STS).
+
+[IMPORTANT]
+====
+It is not supported to use the {secrets-store-operator} with {secrets-store-provider} in a hosted control plane cluster.
+====
+
+.Prerequisites
+
+* Your cluster is installed on AWS and uses AWS Security Token Service (STS).
+* You have installed the {secrets-store-operator}. See _Installing the {secrets-store-driver}_ for instructions.
+* You have configured {secrets-store-provider} to store the required secrets.
+* You have extracted and prepared the `ccoctl` binary.
+* You have installed the `jq` CLI tool.
+* You have access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Install the {secrets-store-provider} provider:
+
+.. Create a YAML file with the following configuration for the provider resources:
++
+[NOTE]
+====
+This configuration is modified from the configuration provided in the link:https://github.com/aws/secrets-store-csi-driver-provider-aws#installing-the-aws-provider[AWS documentation] so that it works properly with {product-title}.
```

Comment:
@sjstout what is the docs position on linking to partner/integrator upstream docs? 
I don't like this (its yet another github link in our docs); but for the purpose this servers; Vendor B modeled the work they did off of Vendor A; I don't think this is bad (its just not necessarily we need to be in the business of communicating).  These are implementation details the vendor should provide in the vendors docs (correct?).

Reply:
@sferich888 This one was a bit hard. I would normally agree to try to keep as much stuff in the vendor docs as possible (we do make people consult vendor docs for setting up their external secrets store).

But for actually installing the provider config into OCP, we had these sets of vendor docs:

* https://azure.github.io/secrets-store-csi-driver-provider-azure/docs/getting-started/installation/#using-deployment-yamls
* https://github.com/aws/secrets-store-csi-driver-provider-aws#installing-the-aws-provider

Both of them are actually hosted on GitHub too. Plus, the command/configs aren't actually correct to apply as they are shown there. We don't want to users to install into the kube-system namespace. We needed to change that to openshift-cluster-csi-drivers for OCP. Among a few other changes.

The other option would be to link to these above vendor "docs" (which, again, are on GitHub too anyways), but them tell them to not actually follow them as is, because they have to change some of the config first. Plus then we're relying on their YAML, which could change at any time.

If it helps, this is tech preview only. And I do believe the plan in the future is for AWS/Azure to provider these as operators (instead of this installation method). @anjaltelang please correct me if I am misremembering this detail.

Reply:
@sferich888 I think it is quite okay to provide the github links. Reasons below:
1. E-2-e solution requires the Provider install. Currently the installation options are available via the Github links which are also linked off of the SSCSI Upstream documentation. When these Vendors have either Operators or Certified Helm Charts in OpenShift we can link off of their documentation. 
2. This is a TechPreview feature, so having upstream links is fine.  When we GA the feature, we would like downstream/Vendor-specific docs.

Reply:
@bergerhoffer I think your making my point, by linking to vendor doc's (for generic kuberneties cluster) we create situation for OCP customers where in they have a bad or sub-par experience. So while the 'expedient' (fine) thing is to do is, as @anjaltelang suggests, is to link to upstream docs; and claim this is Technology Preivew; if the first feedback we get (from our preview) is the instructions don't work (couldn't evaluate), what good was the preview effort? 

The no github links rule exists; because the potential exists for the linked docs to change; breaking existing clusters (customers) where we claim support (with us having little to no recourse to course correct). 

If your intention is to have downstream/Vendor-specific docs; I think we should start that effort in earnest.

Reply:
@sferich888 We are not linking there for them to use it. We pulled the config into the OCP docs for that exact reason - to not allow something that changes in those vendor docs/config to break our OCP customers. We are only linking to give credit/history to where config originated from.

Reply:
Continued conversation via slack. Updated PR with additional notice that the providers are upstream. Got acks via slack afterward